### ISSUE TITLE
IsSecureConnection is not retained w/ Pooling

### ISSUE THREAD

**Issue by iamcarbon** [mysql-net/MySqlConnector#269]
When using the new mysql_clear_password functionality w/ pooling, the second request throws: 

```
MySql.Data.MySqlClient.MySqlException : Authentication method 'mysql_clear_password' requires a secure connection.
```

Without digging deeper, it looks like we may be loosing the SSL mode when getting a connection back from the pool. 

Everything works as expected on the first request and when pooling is explicitly disabled.

Here's the connection string:

```
Server=server;Port=3306;Database=database;User Id=testing;Password=""iamtoken"";SSL Mode=Required;AllowUserVariables=True
```

**Comment by bgrainger** [comment]
Can you post the full call stack for the exception?

**Comment by bgrainger** [comment]
Fixed in [0.20.1](https://github.com/mysql-net/MySqlConnector/releases/tag/0.20.1).
